features/userprofile/impl/src/main/kotlin/io/element/android/features/userprofile/impl/DefaultUserProfilePresenterFactory.kt

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package io.element.android.features.userprofile.impl
+
+import com.squareup.anvil.annotations.ContributesBinding
+import io.element.android.features.userprofile.api.UserProfileState
+import io.element.android.features.userprofile.api.UserProfilePresenterFactory
+import io.element.android.features.userprofile.impl.root.UserProfilePresenter
+import io.element.android.libraries.architecture.Presenter
+import io.element.android.libraries.di.SessionScope
+import io.element.android.libraries.matrix.api.core.UserId
+import javax.inject.Inject
+
+@ContributesBinding(SessionScope::class)
+class DefaultUserProfilePresenterFactory @Inject constructor(
+    private val factory: UserProfilePresenter.Factory,
+) : UserProfilePresenterFactory {
+    override fun create(userId: UserId): Presenter<UserProfileState> = factory.create(userId)
+}
```

Comment:
This double factory level is a bit weird, there is maybe a better way to handle this. Any ideas @ganfra @jmartinesp ?

Reply:
I can't see any, sorry... I was going to suggest trying to reuse `UserProfilePresenter.Factory` but that's not possible since the `AssistedFactory` needs to return a concrete type.

The best I could get was making `UserProfilePresenter.Factory` implement `UserProfilePresenterFactory`, then use it in `DefaultUserProfilePresenterFactory` like this:

```kotlin
@ContributesBinding(SessionScope::class)
class DefaultUserProfilePresenterFactory @Inject constructor(
    private val factory: UserProfilePresenter.Factory,
) : UserProfilePresenterFactory by factory
```

Reply:
Thanks for looking into it. Let's keep it like this then!